client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx

```diff
@@ -51,6 +51,37 @@ interface IAnalysisWizard {
   isOpen: boolean;
 }
 
+const determineMode = (
+  applications: Application[]
+):
+  | "binary"
+  | "source-code-deps"
+  | "source-code"
+  | "binary-upload"
+  | undefined => {
+  // If only one application is selected
+  if (applications.length === 1) {
```

Comment:
Please provide some reference/description for the algorithm (issue on GH?)

Reply:
We added a function according to the conditions requested in the issue
And there are comments in the code explaining how the function works
Attached is a link to the GitHab issue https://github.com/konveyor/tackle2-ui/issues/1837